### PR TITLE
Fixes a bug caused by big numbers on annotations

### LINKED ISSUE #990: [BUG] A big number in a comment causes crash in annotations parser
All you have to do is to add an operator declaration like this:

```tla
\* 100000000000000000000
X == 1
```

The stack trace is like follows:

```
(Please report an issue at: [https://github.com/informalsystems/apalache/issues],java.lang.NumberFormatException: For input string: "100000000000000000000")
Unhandled exception                                               E@12:26:03.447
java.lang.NumberFormatException: For input string: "100000000000000000000"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at scala.collection.immutable.StringLike.toInt(StringLike.scala:304)
	at scala.collection.immutable.StringLike.toInt$(StringLike.scala:304)
	at scala.collection.immutable.StringOps.toInt(StringOps.scala:33)
	at at.forsyte.apalache.io.annotations.parser.AnnotationLexer$.$anonfun$number$1(AnnotationLexer.scala:98)
	at scala.util.parsing.combinator.Parsers$Success.map(Parsers.scala:138)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$map$1(Parsers.scala:258)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$append$2(Parsers.scala:270)
	at scala.util.parsing.combinator.Parsers$Failure.append(Parsers.scala:200)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$append$1(Parsers.scala:270)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$append$1(Parsers.scala:270)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$append$1(Parsers.scala:270)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$append$1(Parsers.scala:270)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers.$anonfun$positioned$1(Parsers.scala:970)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.RegexParsers$$anon$3.apply(RegexParsers.scala:137)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$flatMap$1(Parsers.scala:255)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers.$anonfun$rep1$1(Parsers.scala:796)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$append$1(Parsers.scala:270)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$map$1(Parsers.scala:258)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$Success.flatMapWithNext(Parsers.scala:144)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$flatMap$1(Parsers.scala:255)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$flatMap$1(Parsers.scala:255)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$map$1(Parsers.scala:258)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$Parser.$anonfun$flatMap$1(Parsers.scala:255)
	at scala.util.parsing.combinator.Parsers$$anon$1.apply(Parsers.scala:225)
	at scala.util.parsing.combinator.Parsers$$anon$6.apply(Parsers.scala:986)
	at scala.util.parsing.combinator.RegexParsers.parse(RegexParsers.scala:182)
	at scala.util.parsing.combinator.RegexParsers.parse$(RegexParsers.scala:181)
	at at.forsyte.apalache.io.annotations.parser.AnnotationLexer$.parse(AnnotationLexer.scala:17)
	at scala.util.parsing.combinator.RegexParsers.parseAll(RegexParsers.scala:190)
	at scala.util.parsing.combinator.RegexParsers.parseAll$(RegexParsers.scala:189)
	at at.forsyte.apalache.io.annotations.parser.AnnotationLexer$.parseAll(AnnotationLexer.scala:17)
	at at.forsyte.apalache.io.annotations.parser.AnnotationLexer$.apply(AnnotationLexer.scala:30)
	at at.forsyte.apalache.io.annotations.AnnotationParser$.parse(AnnotationParser.scala:105)
	at at.forsyte.apalache.io.annotations.AnnotationParser$.parse(AnnotationParser.scala:120)
	at at.forsyte.apalache.tla.imp.AnnotationExtractor.parseAndSave(AnnotationExtractor.scala:18)
	at at.forsyte.apalache.tla.imp.OpDefTranslator.translate(OpDefTranslator.scala:72)
	at at.forsyte.apalache.tla.imp.ModuleTranslator.$anonfun$translate$4(ModuleTranslator.scala:70)
	at at.forsyte.apalache.tla.imp.ModuleTranslator.$anonfun$translate$4$adapted(ModuleTranslator.scala:62)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at at.forsyte.apalache.tla.imp.ModuleTranslator.translate(ModuleTranslator.scala:62)
	at at.forsyte.apalache.tla.imp.SanyImporter.$anonfun$loadFromFile$1(SanyImporter.scala:46)
	at scala.collection.IndexedSeqOptimized.foldLeft(IndexedSeqOptimized.scala:60)
	at scala.collection.IndexedSeqOptimized.foldLeft$(IndexedSeqOptimized.scala:68)
	at scala.collection.mutable.ArrayOps$ofRef.foldLeft(ArrayOps.scala:198)
	at at.forsyte.apalache.tla.imp.SanyImporter.loadFromFile(SanyImporter.scala:44)
	at at.forsyte.apalache.tla.imp.passes.SanyParserPassImpl.execute(SanyParserPassImpl.scala:61)
	at at.forsyte.apalache.infra.passes.PassChainExecutor.exec$1(PassChainExecutor.scala:22)
	at at.forsyte.apalache.infra.passes.PassChainExecutor.run(PassChainExecutor.scala:37)
	at at.forsyte.apalache.tla.Tool$.runCheck(Tool.scala:187)
	at at.forsyte.apalache.tla.Tool$.$anonfun$run$3(Tool.scala:95)
	at at.forsyte.apalache.tla.Tool$.handleExceptions(Tool.scala:322)
	at at.forsyte.apalache.tla.Tool$.run(Tool.scala:95)
	at at.forsyte.apalache.tla.Tool$.main(Tool.scala:45)
	at at.forsyte.apalache.tla.Tool.main(Tool.scala)
```

This is a funny issue. @shonfeder or @rodrigo7491, would you like to try to fix this one?

### PR DESCRIPTION
**Pull request by rodrigo7491**
- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] ~~Documentation added for any new functionality~~
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

Fixes a bug caused by big numbers on annotations. Closes #990.

### CODE REVIEW

File: tla-io/src/main/scala/at/forsyte/apalache/io/annotations/parser/AnnotationLexer.scala [hunk 1]
```diff
@@ -95,7 +95,7 @@ object AnnotationLexer extends RegexParsers {
   }
 
   private def number: Parser[NUMBER] = {
-    "(|-)[0-9][0-9]*".r ^^ { value => NUMBER(value.toInt) }
+    "(|-)[0-9][0-9]*".r ^^ { value => NUMBER(BigInt(value)) }
   }
```
Comment (shonfeder) [inline tla-io/src/main/scala/at/forsyte/apalache/io/annotations/parser/AnnotationLexer.scala:99]: Why do we parse out random numbers in comments at all? Shouldn't we only proceed with parsing annotations if they begin with the `@` indicator? @konnov

**Review by shonfeder — COMMENTED** [review]
This looks good to me as a fix (awesome work), but have a question around why we are parsing integers in comments at all. It's not clear to me why a random integer in a comment should be taken as an annotation.

File: tla-io/src/main/scala/at/forsyte/apalache/io/annotations/parser/AnnotationLexer.scala (hunk 1 above)
Comment (konnov) [reply tla-io/src/main/scala/at/forsyte/apalache/io/annotations/parser/AnnotationLexer.scala:99]: That's a good point. In the earlier versions, we were looking for `@foo(...)`, but that somehow did not work with multiline strings that were starting with `\*`. So we needed a proper lexer. And now we have this issue that the lexer was not accepting a big integer. The challenge here is how to filter out an annotation from the sea of arbitrary characters.

**Review by konnov — APPROVED** [review]
Thanks for the fix! It looks great.

We have to think about the question raised by @shonfeder . But that should probably go in a separate issue.

**Review by shonfeder — APPROVED** [review]

**Comment by konnov** [comment]
as an aftermath of this PR, it looks like https://github.com/informalsystems/apalache/blob/unstable/tla-types/src/test/scala/at/forsyte/apalache/tla/typecheck/TestDefaultType1Parser.scala is sometimes running out of memory on Scalacheck tests

**Pull request merged**